### ISSUE TITLE
MissingRequiredPropertyException: Missing required property 'ErrorCause.reason'

### ISSUE THREAD

**Issue by afimuc** [elastic/elasticsearch-java#339]
### Java API client version

7.17.3

### Java version

13

### Elasticsearch Version

7.17.5

### Problem description

Sending requests using `co.elastic.clients.elasticsearch.ElasticsearchAsyncClient#bulk(co.elastic.clients.elasticsearch.core.BulkRequest)`

An error during the processing of the request leads to
```
Error deserializing co.elastic.clients.elasticsearch._types.ErrorCause: co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'ErrorCause.reason' (JSON path: items[1].error.caused_by.caused_by.caused_by) (line no=1, column no=953, offset=-1)
at co.elastic.clients.json.JsonpMappingException.from0(JsonpMappingException.java:134) ~[elasticsearch-java-7.17.3.jar:na]
...
Caused by: co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'ErrorCause.reason'
	at co.elastic.clients.util.ApiTypeHelper.requireNonNull(ApiTypeHelper.java:76) ~[elasticsearch-java-7.17.3.jar:na]
	at co.elastic.clients.elasticsearch._types.ErrorCause.<init>(ErrorCause.java:81) ~[elasticsearch-java-7.17.3.jar:na]
	at co.elastic.clients.elasticsearch._types.ErrorCause.<init>(ErrorCause.java:56) ~[elasticsearch-java-7.17.3.jar:na]
	at co.elastic.clients.elasticsearch._types.ErrorCause$Builder.build(ErrorCause.java:375) ~[elasticsearch-java-7.17.3.jar:na]
	at co.elastic.clients.elasticsearch._types.ErrorCause$Builder.build(ErrorCause.java:216) ~[elasticsearch-java-7.17.3.jar:na]
	at co.elastic.clients.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:80) ~[elasticsearch-java-7.17.3.jar:na]
	at co.elastic.clients.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:43) ~[elasticsearch-java-7.17.3.jar:na]
	at co.elastic.clients.json.ObjectDeserializer$FieldObjectDeserializer.deserialize(ObjectDeserializer.java:71) ~[elasticsearch-java-7.17.3.jar:na]
	at co.elastic.clients.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:180) ~[elasticsearch-java-7.17.3.jar:na]
	... 52 common frames omitted
```

**Comment by xiaohuan666** [comment]
@swallez  hello,I had the same problem,what should i do?  Upgrade es version? or Upgrade java client version? which version should i upgrade

**Comment by swallez** [comment]
@xiaohuan666 the fix will be part of versions 8.4.2 that will be released in the next 2 weeks and 7.17.7 that should be released in october.

In the meantime, if your environment permits, you can also use a [snapshot release](https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/installation.html) for these versions.
